Makefile

```diff
@@ -13,6 +13,10 @@ export GO15VENDOREXPERIMENT=1
 # doesn't allow +, so we use -.
 VERSION := 0.0.1-$(shell date "+%Y%m%d%H%M%S")
 
+DEV_ENV_IMAGE := quay.io/deis/go-dev:0.1.0
+DEV_ENV_WORK_DIR := /go/src/github.com/deis/${SHORT_NAME}
+DEV_ENV_CMD := docker run --rm -v ${PWD}:${DEV_ENV_WORK_DIR} -w ${DEV_ENV_WORK_DIR} ${DEV_ENV_IMAGE}
```

Comment:
Apologies for the long read:

I like that principle in theory, but I think it changes the scope of practical possible usages of clean room. I've seen people build Go binaries in docker containers for one or more of the following reasons:

1. To quickly build a binary for a different OS, where `GOOS`/`GOARCH` aren't enough for their needs. Someone may do this possibly to build against a libc alternative (like musl in alpine) or using a different C compiler (like clang) for cgo dependencies
2. To provide "clean room" environments, like you said
3. To encapsulate all the tools necessary to run builds (`golint`, `goimports`, `make`, etc...)

(1) and (2) collide because non-trivial builds will take long enough to disrupt the edit-test cycle. And we can get (3) with both other options.

I prefer (1) for speed reasons but agree that clean room builds are important. Maybe we could provide a `clean-build` target so that developers can periodically check their dependencies are still in order? Such a target would be especially useful as it mimics our CI builds.